go/protocols/ops/stats_test.go

```diff
@@ -0,0 +1,36 @@
+package ops
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/bradleyjkemp/cupaloy"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_StatsSerde_RoundTrip(t *testing.T) {
+	want := &Stats_DocsAndBytes{
+		DocsTotal:            1,
+		BytesTotal:           10,
+		XXX_NoUnkeyedLiteral: struct{}{},
+		XXX_unrecognized:     nil,
+		XXX_sizecache:        0,
+	}
+
+	var buf bytes.Buffer
+	require.NoError(t, (&jsonpb.Marshaler{}).Marshal(&buf, want))
+
+	got := new(Stats_DocsAndBytes)
+	require.NoError(t, (&jsonpb.Unmarshaler{}).Unmarshal(bytes.NewReader(buf.Bytes()), got))
```

Comment:
Does the jsonpb unmarshaler accept either a string-integer or integer representation?

Reply:
It does accept both. It is not surprising that the string-integer version works since that how it gets marshaled by default, which we are overriding with the `MarshalJSONPB` implementation. This test shows that we don't also need a custom `UnmarshalJSONPB` implementation to handle the integer representation, since it already works with `jsonpb.Unmarshaler`.